yaptide/celery/utils/manage_tasks.py

```diff
@@ -87,3 +92,54 @@ def translate_celery_state_naming(job_state: str) -> str:
         return EntityState.COMPLETED.value
     # Others are the same
     return job_state
+
+
+def handle_shieldhit_cancellation(tasks: list[CeleryTaskModel]):
+    """Function cancelling shieldhit processes"""
+    command_as_list = ['kill', '--signal', 'SIGINT']
+    celery_ids = []
+    for task in tasks:
+        if task.task_state == EntityState.RUNNING.value:
+            command_as_list.append(str(task.sim_pid))
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.sim_pid is None:
+            celery_ids.append(task.celery_id)
+
+    # Send SIGINT to Shieldhit processes
+    subprocess.run(command_as_list, check=True)
+    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+
+
+def handle_fluka_cancellation(tasks: list[CeleryTaskModel]):
+    """Function cancelling fluka processes"""
+    FILE_NAME = 'rfluka.stop'
+    celery_ids = []
+    for task in tasks:
+        if task.task_state == EntityState.RUNNING.value:
+            try:
+                if (target_folder := next(Path(task.path_to_sim).glob("fluka_*/"), None)):
+                    (target_folder / FILE_NAME).write_text("")
+            except OSError as e:
+                logging.warning("Error due to file operation: %s", str(e))
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.path_to_sim is None:
+            celery_ids.append(task.celery_id)
+    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
```

Comment:
maybe you should wait till results are really written to the disk before terminating all Fluka tasks ?
if you terminate task while in the process of writing the output you will get corrupted file.
if I remember correctly Fluka would notify you on writing the results by removing the `rfluka.stop` file.

Reply:
It terminates only workers that don't start simulation before user click cancel button. Descripted case don't occurs.

Reply:
OK, I've got it now. So indeed this requires very good comments.
You needs to say explicitely that you ask Fluka to dump results and stop by creating `rfluka.stop` file. The comment should say that you deliberately DO NOT revoke such tasks, cause you need to wait till it dump results to disk (which may take some time, couple of seconds), and you expect that this tasks will exit by itself.